### PR TITLE
Add Docker for R notebooks

### PR DESCRIPTION
**Pull request by trang1618**

### CODE REVIEW

File: environment/r/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM rocker/geospatial:latest
+WORKDIR /home/rstudio
+RUN apt-get update \
+    && apt-get -y install libgsl0-dev \
+    && apt -y install libxml2-dev
+# install R packages
+RUN install2.r lubridate wru rnaturalearth epitools caret cowplot
+
+EXPOSE 8787
```
Comment (dhimmel) [inline environment/r/Dockerfile:9]: missing newline

File: readme.md [hunk 2]
```diff
@@ -51,14 +51,30 @@ docker run \
   --mount type=bind,source="$(pwd)",target=/user/jupyter \
   docker.pkg.github.com/greenelab/iscb-diversity/iscb-diversity
 ```
-
```
Comment (dhimmel) [inline readme.md:54]: blank lines around code blocks are customary

File: 092.save-raws-to-Rdata.Rmd [hunk 3]
```diff
@@ -13,18 +13,25 @@ knit: (function(inputFile, encoding) {
...
-library(rnaturalearth)
 source('utils/r-utils.R')
 theme_set(theme_bw() + theme(legend.title = element_blank()))
 ```
 
 
+```{r}
+# library(rnaturalearth)
+# world <- rnaturalearth::ne_countries(scale='medium', returnclass = 'sf')
+# save(world, file = 'Rdata/world.Rdata')
```
Comment (dhimmel) [inline 092.save-raws-to-Rdata.Rmd:24]: Maybe add a comment that rnaturalearth does not work in the docker  (why does it not work?) and that is why we're loading it from Rdata

File: .github/workflows/docker-r.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
+name: Build & Publish Docker
```
Comment (dhimmel) [inline .github/workflows/docker-r.yaml:1]:
```suggestion
name: Build & Publish R Docker
```

File: .github/workflows/docker-r.yaml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+on:
+  push:
+    paths:
+    - environment/r/Dockerfile
+    - environment/r/requirements.txt
+    - .github/workflows/docker-r.yaml
+jobs:
+  docker:
+    name: Build & Publish Docker
```
Comment (dhimmel) [inline .github/workflows/docker-r.yaml:10]:
```suggestion
    name: Build & Publish R Docker
```

**Review by dhimmel — APPROVED** [review]
looks good besides a few minor comments.

once merged we'll see whether it deploys properly to the github package registry

File: 092.save-raws-to-Rdata.Rmd (hunk 3 above)
Comment (trang1618) [reply 092.save-raws-to-Rdata.Rmd]: managed to make it work!

**Pull request merged**